asserts/headers.go

```diff
@@ -106,11 +86,18 @@ func parseEntry(consumedByIntro int, first int, lines []string, baseIndent int)
 	i := first + 1
 	if consumedByIntro == len(entry) {
 		// multiline values
-		if i < len(lines) && strings.HasPrefix(lines[i], nestingPrefix(baseIndent, listPrefix)) {
-			// list
-			return parseList(i, lines, baseIndent)
+		basePrefix := nestingPrefix(baseIndent, commonPrefix)
+		if i < len(lines) && strings.HasPrefix(lines[i], basePrefix) {
+			rest := lines[i][len(basePrefix):]
+			if strings.HasPrefix(rest, listChar) {
+				// list
+				return parseList(i, lines, baseIndent)
```

Comment:
Do we ever look back in the list?  If we don't, it sounds like we're reimplementing the internal indexing of slices externally. We might pass lines[i:] in and lines[firstAfter:] out, and drop the indexing both ways.